script.js

```diff
@@ -0,0 +1,160 @@
+// @ts-check
+
+import { composeCreateOrUpdateTextFile } from "@octokit/plugin-create-or-update-text-file";
+import prettier from "prettier";
+import YAML from 'yaml'
+
+const { parseDocument, parse, stringify, Document } = YAML
+
+const isYamlFile = (fileName) => {
+  const yamlFilePattern = /^.*\.ya?ml$/;
+  return yamlFilePattern.test(fileName);
+};
+
+/**
+ * Add cache parameter to GitHub Actions using setup-node
+ *
+ * @param {import('@octoherd/cli').Octokit} octokit
+ * @param {import('@octoherd/cli').Repository} repository
+ * @param {object} options
+ * @param {string} [options.cache] Select which package manager you want to use for caching
+ */
+export async function script(octokit, repository, { cache = "npm" }) {
+  // Global variables used throughout the code
+  const owner = repository.owner.login;
+  const repo = repository.name;
+  const defaultBranch = repository.default_branch;
+  const branchName = "add-cache-to-node-workflows";
+  const path = ".github/workflows";
+
+  // Get info on repository branches
+  const { data: branches } = await octokit.request(
+    "GET /repos/{owner}/{repo}/branches",
+    {
+      owner,
+      repo,
+      branch: defaultBranch,
+    }
+  );
+
+  // Get SHA of repository's default branch
+  const sha = branches
+    .filter((branch) => branch.name === defaultBranch)
+    .map((branch) => branch.commit.sha)[0];
+  const branchExists = branches.some((branch) => branch.name === branchName);
+
+  // Create branch if not present
+  if (!branchExists) {
+    const ref = await octokit
+      .request("POST /repos/{owner}/{repo}/git/refs", {
+        owner,
+        repo,
+        ref: `refs/heads/${branchName}`,
+        sha,
+      })
+      .then((response) => response.data.ref);
+
+    if (!ref) {
+      octokit.log.warn(`Error creating branch in ${repository.html_url}`);
+
+      return;
+    }
```

Comment:
Is there the possibility that the request succeeds but the `ref` property is not set?

Reply:
> Is there the possibility that the request succeeds but the `ref` property is not set?

I copy-pasted it from [wolfy's script](https://github.com/wolfy1339/octoherd-script-fix-prettier-update/blob/9861452e6e60363c2cfeba823273ef7cec74b653/script.js#L48). SInce it is supposed to return a string or throw I will remove it and let's see if it gives problems